iothub/device/src/Transport/AmqpIot/AmqpIotTransport.cs

```diff
@@ -75,17 +71,26 @@ internal async Task<TransportBase> InitializeAsync(CancellationToken cancellatio
                 Logging.Enter(this, nameof(InitializeAsync));
 
             TransportBase transport;
+            AmqpTransportInitiator amqpTransportInitiator;
 
             switch (_amqpTransportSettings.Protocol)
             {
                 case IotHubClientTransportProtocol.Tcp:
-                    var amqpTransportInitiator = new AmqpTransportInitiator(_amqpSettings, _tlsTransportSettings);
+                    amqpTransportInitiator = new AmqpTransportInitiator(_amqpSettings, _tlsTransportSettings);
                     transport = await amqpTransportInitiator.ConnectAsync(cancellationToken).ConfigureAwait(false);
                     break;
 
                 case IotHubClientTransportProtocol.WebSocket:
-                    transport = _clientWebSocketTransport = (ClientWebSocketTransport)await CreateClientWebSocketTransportAsync(cancellationToken)
-                        .ConfigureAwait(false);
+                    var websocketUri = new Uri($"{Scheme}{_hostName}:{SecurePort}{UriSuffix}");
+                    var websocketTransportSettings = new WebSocketTransportSettings
+                    {
+                        Uri = websocketUri,
+                        Proxy = _amqpTransportSettings.Proxy,
+                        SubProtocol = Amqpwsb10,
+                    };
+
+                    amqpTransportInitiator = new AmqpTransportInitiator(_amqpSettings, websocketTransportSettings);
+                    transport = await amqpTransportInitiator.ConnectAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
This line can be moved outside of the switch statement since we execute it regardless of WS vs TCP